src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -283,16 +325,26 @@ class HuiMapCard extends LitElement implements LovelaceCard {
       return;
     }
 
-    const bounds = this.Leaflet.latLngBounds(
-      this._mapItems ? this._mapItems.map((item) => item.getLatLng()) : []
-    );
+    const bounds = this.Leaflet.featureGroup(this._mapItems).getBounds();
     this._leafletMap.fitBounds(bounds.pad(0.5));
 
     if (zoom && this._leafletMap.getZoom() > zoom) {
       this._leafletMap.setZoom(zoom);
     }
   }
 
+  private _getColor(name: string) {
+    let color;
+    if (this._colorDict[name]) {
+      color = this._colorDict[name];
+    } else {
```

Comment:
This is not possible since we use entity IDs in this function only.

Reply:
@balloob What do you mean by this is not possible? 
The dict is to ensure same entitys get the same color for marker and path. Are entity IDs not uniqe?

Reply:
My bad. I didn't realize it was a caching function. This is ok.